### ISSUE TITLE
NexusParser support for state alphabets longer than 10 and Equate symbols

### ISSUE THREAD

**Issue by jjmccollum** [CompEvol/beast2#1050]
Rob Turnbull (@rbturnbull) and I are using BEAST2 for a phylogenetic analysis of a textual tradition, and one of our recent efforts in this project has been the development of a general-purpose tool (https://github.com/jjmccollum/teiphy) for converting manuscript collations in TEI XML format (a standard in the digital humanities) to formats like NEXUS for phylogenetic analysis. We can now generate NEXUS output that works with several phylogenetic programs, but when I try to import the output NEXUS file in BEAUti 2, I get the following error:

```
java.lang.NumberException: For input string "b"
```

This appears to be caused by the presence of a non-numeric state symbol in the first line of my NEXUS file's matrix block:

```
Format
    DataType=Standard
    Symbols="0 1 2 3 4 5 6 7 8"
    Equate="9=(01) a=(0124) b=(02) c=(03) d=(04) e=(046) f=(05) g=(07) h=(34)";
Matrix
    P46                    2017210?001433210601001001001104700b0302
```

It would be helpful for our purposes to extend the `NexusParser` class's support for the Standard datatype (following #157) to allow for alphabetical characters as symbols, since some points of variation (resp. characters) in a textual collation will have more than ten substantive variant readings (resp. states). If the singleton state symbols are defined in the Symbols block (also shown in the example above), then the `NexusParser` could read them from this block before it parses the matrix, if that simplifies things.

In addition, support for additional symbols in the Equate block (which defines convenient one-character symbols for ambiguous states so that the sequences in the matrix remain nicely aligned), would be a useful extension of support for NEXUS inputs.

**Comment by rbouckaert** [comment]
That looks like a useful extension. It might only need some adjustment of the [`Standard`](https://github.com/CompEvol/beast2/blob/master/src/beast/base/evolution/datatype/StandardData.java) data type, which takes care of encodings.

Do I understand it correctly that a `b` in the example represents the ambiguous character `0` or `2`? If so, this can currently be handled by inserting `{02}` in the sequence instead of `b`. Equate block support would be more elegant.

**Comment by jjmccollum** [comment]
Yes, the `b` in the example is equivalent to inserting `{02}` in the matrix; at the expense of using more distinct characters, the Equate block should allow users to include ambiguous characters while maintaining nicely aligned sequences in the matrix.

But I should clarify that the `java.lang.NumberException` will be raised even if there is no equate block and we had something like

```
DataType=Standard
    Symbols="0 1 2 3 4 5 6 7 8 9 a b"
```

The most important extension would be support for a symbols alphabet beyond the digits 0-9.
